components/Buttons/ButtonAddToBrowser.tsx

```diff
@@ -21,18 +21,27 @@ export default function ButtonAddToBrowser() {
 
   useEffect(() => {
     if (isBrowser && isChrome) {
-      setBrowserName('chrome')
       /* eslint-disable-next-line no-undef */
-      chrome.runtime.sendMessage(config.CHROME_EXTENSION_ID, {
-        action: 'id',
-        value: config.CHROME_EXTENSION_ID,
-      })
+      chrome.runtime.sendMessage(
+        config.CHROME_EXTENSION_ID,
+        {
+          action: 'id',
+          value: config.CHROME_EXTENSION_ID,
+        },
+        function (response) {
+          if (!response) {
+            return setBrowserName('chrome')
+          }
+        }
+      )
     }
 
     if (isFirefox) {
-      setBrowserName('firefox')
+      if (!window.extensionInterface) {
+        setBrowserName('firefox')
+      }
     }
-  }, [])
+  }, [browserName])
```

Comment:
I don't think this dependency should be here. `browserName` can only change within this effect. We don't want to run it again because of this change. Just once after component mount (`[]`) should be fine.

Reply:
uh-oh forgot to remove that, thanks!